r2i/onnxrt/parameters.cc

```diff
@@ -0,0 +1,222 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include "parameters.h"
+
+#include <string>
+
+namespace r2i {
+namespace onnxrt {
+
+#define PARAM(_name, _desc, _flags, _type, _acc) \
+  {						 \
+    (_name),					 \
+    {						 \
+      .meta = {					 \
+	.name = (_name),			 \
+	.description = (_desc),			 \
+	.flags = (_flags),			 \
+	.type = (_type),			 \
+      },					 \
+      .accessor = (_acc)			 \
+    }						 \
+  }
+
+Parameters::Parameters () :
+  parameter_map ({
+  /* Engine parameters */
+  PARAM("logging-level", "ONNXRT Logging Level",
+        r2i::ParameterMeta::Flags::READWRITE | r2i::ParameterMeta::Flags::WRITE_BEFORE_START,
```

Comment:
How is this working? On the engine you expect this parameter to be set on the STARTED state, but here you expect it to be set before STARTED.

Reply:
Do you really need that flag? if not, remove it.